File: frontend-app/src/app/services/puesto-empresa.service.ts
import { HttpClient, HttpParams } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { PuestoEmpresa } from "../domain/PuestoEmpresa";

@Injectable({
    providedIn: 'root'
})
export class PuestoEmpresaService{

    private url:string = 'http://localhost:8080/api/puestos-empresas';

    constructor(private http: HttpClient){

    }

    listadoPuestoEmpresas(): Observable<PuestoEmpresa[]>{
        return this.http.get<PuestoEmpresa[]>(this.url);
    }
        
    buscarPuestoEmpresaPorId(id: number): Observable<PuestoEmpresa>{
        return this.http.get<PuestoEmpresa>(`${this.url}/${id}`);
    }

    guardarPuestoEmpresa(puestoEmpresa: PuestoEmpresa) {
        return this.http.post(`${this.url}/guardar`, puestoEmpresa);
    }
    
    actualizarPuestoEmpresa(puestoEmpresa: PuestoEmpresa) {
        return this.http.put(`${this.url}/actualizar`, puestoEmpresa);
    }
    
    eliminarPuestoEmpresa(idPuestoEmpresa: number){
        return this.http.delete(`${this.url}/${idPuestoEmpresa}`);
    }

    
    listadoPuestoEmpresaPaginacion(page: number): Observable<any>{
        return this.http.get<any[]>(`${this.url}/page/${page}`);
    }   

    buscarPuestoEmpresa(puestoEmpresa: string): Observable<PuestoEmpresa[]>{
        const params = new HttpParams().set('puestoEmpresa', puestoEmpresa);

        return this.http.get<PuestoEmpresa[]>(`${this.url}/buscar`, {params});
    }

    listadoPuestosEmpresasCompletos(): Observable<PuestoEmpresa[]>{
        return this.http.get<PuestoEmpresa[]>(`${this.url}/completos`);
    }


}

File: backend-app/src/main/java/com/backend_app/backend_app/service/PuestoEmpresaService.java
package com.backend_app.backend_app.service;

import com.backend_app.backend_app.domain.PuestoEmpresa;
import com.backend_app.backend_app.dto.PuestoEmpresaDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

public interface PuestoEmpresaService {

    List<PuestoEmpresaDTO> listadoPuestosEmpresas();

    Optional<PuestoEmpresaDTO> buscarPuestoEmpresaPorId(long idPuestoEmpresa);

    PuestoEmpresaDTO convertirADTO(PuestoEmpresa puestoEmpresa);

    void insertarPuestoEmpresa(PuestoEmpresaDTO puestoEmpresaDTO);

    void actualizarPuestoEmpresa(PuestoEmpresaDTO puestoEmpresaDTO);

    void eliminarPuestoEmpresa(PuestoEmpresaDTO puestoEmpresaDTO);

    Page<PuestoEmpresaDTO> findAll(Pageable pageable);

    List<PuestoEmpresaDTO> buscarPuestoEmpresa(String puestoEmpresa);

    List<PuestoEmpresaDTO> listadoPuestosEmpresasCompletos();

}
